completions.go

```diff
@@ -592,9 +591,12 @@ You will need to start a new shell for this setup to take effect.
 		DisableFlagsInUseLine: true,
 		ValidArgsFunction:     NoFileCompletions,
 		RunE: func(cmd *Command, args []string) error {
-			return cmd.Root().GenBashCompletion(out)
+			return cmd.Root().GenBashCompletionV2(out, !noDesc)
```

Comment:
Just so I understand correctly, we are now by _default_ using the new v2 version of Bash completions? Can users expect any kind of changes in behavior with this new version when using the generated completion command?

I think I've mentioned this before, but I just want to be extra careful about shipping breaking changes.

Reply:
So the default completion command will use bash completion v2, but any projects that already provide their own 'completion' command will keep using V1 until they choose to change it themselves.

The main differences between v1 and v2 are:
- no more suggesting the `--flag=` form
- completion descriptions will appear by default (these are not available in v1)

One special case will be projects that provide completion through a different command name (say a command named "complete").  Those projects will continue to use v1 for their own command but will also provide cobra's implicit "completion" command which will use v2, unless of course, they take the time to disable the default "completion" command.

That special situation would benefit from a mention in the release notes, I think.